source/h1_stream.c

```diff
@@ -4,103 +4,174 @@
  */
 #include <aws/http/private/h1_stream.h>
 
-#include <aws/http/private/connection_impl.h>
+#include <aws/http/private/h1_connection.h>
+#include <aws/http/private/h1_encoder.h>
 
 #include <aws/http/status_code.h>
 #include <aws/io/logging.h>
 #include <aws/io/stream.h>
 
+#include <inttypes.h>
+
 static void s_stream_destroy(struct aws_http_stream *stream_base) {
     struct aws_h1_stream *stream = AWS_CONTAINER_OF(stream_base, struct aws_h1_stream, base);
+    AWS_ASSERT(
+        stream->synced_data.api_state != AWS_H1_STREAM_API_STATE_ACTIVE &&
+        "Stream should be complete (or never-activated) when stream destroyed");
+    AWS_ASSERT(
+        aws_linked_list_empty(&stream->thread_data.chunk_list) &&
+        aws_linked_list_empty(&stream->synced_data.chunk_list) &&
+        "Chunks should be marked complete before stream destroyed");
 
-    aws_h1_stream_body_chunks_clean_up(stream);
     aws_h1_encoder_message_clean_up(&stream->encoder_message);
     aws_byte_buf_clean_up(&stream->incoming_storage_buf);
     aws_mem_release(stream->base.alloc, stream);
 }
 
-static void s_stream_update_window(struct aws_http_stream *stream, size_t increment_size) {
-    aws_http_connection_update_window(stream->owning_connection, increment_size);
+static struct aws_h1_connection *s_get_h1_connection(const struct aws_h1_stream *stream) {
+    return AWS_CONTAINER_OF(stream->base.owning_connection, struct aws_h1_connection, base);
+}
+
+static void s_stream_lock_synced_data(struct aws_h1_stream *stream) {
+    aws_h1_connection_lock_synced_data(s_get_h1_connection(stream));
 }
 
-static int s_body_chunks_init(struct aws_h1_stream *stream) {
```

Comment:
These diffs are a mess.
A bunch of chunk initialization code was removed from h1_stream.c
the new code is unrelated